### ISSUE TITLE
Issue when upgrading from 0.11.3 to 0.12.0 that persists through 1.4.0

### ISSUE THREAD

**Issue by soluml** [webpack-contrib/mini-css-extract-plugin#733]
- Operating System: Mac OS
- Node Version:  13.11.0
- NPM Version: 6.14.8
- webpack Version: 5.28.0
- mini-css-extract-plugin Version: 1.4.0

### Expected Behavior

That I can build successfully. For some reason I've been unable to get mini-css-extract-plugin to work with Webpack 5. When I use style-loader, everything works fine, but if I set the mode to production and use this plugin, the build blows up with the below. It fails consistently, regardless of what's in the css/scss files. Even an empty main.css file imported via entry blows up with the following error.

There are no issues if I move back to 0.11.3. If I move to 0.12.0, the issue starts.  I noticed on the releases page and when [comparing the two versions](https://github.com/webpack-contrib/mini-css-extract-plugin/compare/v0.11.3...v0.12.0) that a new feature was added in v0.12.0: **opt-in to transitive only side effects (webpack@5), no more empty JS chunks**

I'm not sure if this is a feature we can disable or if there is something in the build that 0.11.3 tolerated that 0.12.0 and beyond do not?

Any help would be appreciated, thanks!

### Actual Behavior

```
> webpack --config ./webpack.prod.js && clientlib --verbose

/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/ExportsInfo.js:370
                for (const exportInfo of this._exports.values()) {
                                                       ^

RangeError: Maximum call stack size exceeded
    at MapIterator.[Symbol.iterator] (<anonymous>)
    at ExportsInfo.setUsedWithoutInfo (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/ExportsInfo.js:370:42)
    at processReferencedModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:75:25)
    at processModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:263:9)
    at processModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:197:10)
    at processModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:197:10)
    at processModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:197:10)
    at processModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:197:10)
    at processModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:197:10)
    at processModule (/Users/bsolum/Documents/aem-playground/modified-archetype/ui.frontend/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:197:10)
```

### Code

```js
    rules: [
        {
          test: /\.s?css$/,
          exclude: /node_modules/,
          use: [
            env.dev ? "style-loader" : MiniCssExtractPlugin.loader,
            {
              loader: "css-loader",
            },
            {
              loader: "sass-loader",
            },
          ],
        },
      ],
    },
    plugins: [
      new CleanWebpackPlugin(),
      new MiniCssExtractPlugin({
        filename: "clientlib-[name]/[name].css",
      }),
      new CopyWebpackPlugin({
        patterns: [
          {
            from: path.resolve(__dirname, SOURCE_ROOT + "/resources"),
            to: "./clientlib-site/",
          },
        ],
      }),
    ],
```

```js
  "devDependencies": {
    "@babel/core": "^7.13.10",
    "@babel/eslint-parser": "^7.13.10",
    "@babel/preset-env": "^7.13.12",
    "@typescript-eslint/eslint-plugin": "^4.19.0",
    "@typescript-eslint/parser": "^4.19.0",
    "aem-clientlib-generator": "^1.7.5",
    "aemsync": "^4.0.3",
    "autoprefixer": "^10.2.5",
    "babel-loader": "^8.2.2",
    "browserslist": "^4.16.3",
    "clean-webpack-plugin": "^3.0.0",
    "copy-webpack-plugin": "^8.1.0",
    "css-loader": "^5.2.0",
    "cssnano": "^5.0.0-rc.2",
    "cssnano-preset-advanced": "^4.0.7",
    "eslint": "^7.22.0",
    "eslint-plugin-prettier": "^3.3.1",
    "glob-import-loader": "^1.1.4",
    "mini-css-extract-plugin": "^1.4.0",
    "postcss": "^8.2.8",
    "postcss-loader": "^5.2.0",
    "prettier": "^2.2.1",
    "sass": "^1.32.8",
    "sass-loader": "^11.0.1",
    "style-loader": "^2.0.0",
    "terser-webpack-plugin": "^5.1.1",
    "ts-loader": "^8.0.18",
    "tsconfig-paths-webpack-plugin": "^3.5.1",
    "typescript": "^4.2.3",
    "watch": "^1.0.2",
    "webpack": "^5.28.0",
    "webpack-cli": "^4.5.0",
    "webpack-dev-server": "^3.11.2",
    "webpack-merge": "^5.7.3"
  },
```

### How Do We Reproduce?

In a typical webpack build, use the preceding package.json dev dependencies and try to import a .css/,scss file via the main entry file.

**Comment by alexander-akait** [comment]
> In a typical webpack build, use the preceding package.json dev dependencies and try to import a .css/,scss file via the main entry file.

Not enough to reproduce, please create reproducible test repo.

Also `webpack` versoin and other information missed.

Won't fix.

I will reopen when you fixed it.

Please respect time of other developers, I can't guess on the coffee grounds.

**Comment by kevinolivar** [comment]
Hi there,

sorry to revive this issue but it is still happening on the latest version of mini-css-extract-plugin.

I created a repo https://github.com/kevinolivar/aem-test. This repo is normally part of an Adobe AEM project but you should not need to install JavaSDK nor Maven.

## Actual behavior
After installing the packages just run `npm run prod` and you will get

```
➜  aem-test git:(main) ✗ npm run prod

> aem-maven-archetype@1.0.0 prod
> NODE_ENV=production webpack --config ./webpack.prod.js && clientlib --verbose

/Users/kolivar/aem-test/node_modules/webpack/lib/util/runtime.js:60
exports.forEachRuntime = (runtime, fn, deterministicOrder = false) => {
                         ^

RangeError: Maximum call stack size exceeded
    at exports.forEachRuntime (/Users/kolivar/aem-test/node_modules/webpack/lib/util/runtime.js:60:26)
    at ExportInfo.setUsed (/Users/kolivar/aem-test/node_modules/webpack/lib/ExportsInfo.js:1017:4)
    at ExportsInfo.setUsedWithoutInfo (/Users/kolivar/aem-test/node_modules/webpack/lib/ExportsInfo.js:392:31)
    at processReferencedModule (/Users/kolivar/aem-test/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:80:25)
    at processModule (/Users/kolivar/aem-test/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:270:9)
    at processModule (/Users/kolivar/aem-test/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:206:10)
    at processModule (/Users/kolivar/aem-test/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:206:10)
    at processModule (/Users/kolivar/aem-test/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:206:10)
    at processModule (/Users/kolivar/aem-test/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:206:10)
    at processModule (/Users/kolivar/aem-test/node_modules/webpack/lib/FlagDependencyUsagePlugin.js:206:10)
```

I also noticed that changing the mode from 'production' to 'development' was fine in the webpack.prod.js.

## Expected behavior
It should not throw error and finish building same as `npm run dev` (this command is fine). Both, webpack.dev.js and webpack.prod.js use webpack.common.js which includes the MiniCssExtractPlugin loader.

## Environment
OS: Mac OS Big Sur 11.5.2 (20G95)
Node: 16.11.1
NPM : 8.0.0
Yarn: 1.22.5

Hope that helps to investigate and fix that issue.

Thank you very much!

**Comment by alexander-akait** [comment]
@kevinolivar Can you open an issue in webpack, looks like a bug inside webpack code

**Comment by kevinolivar** [comment]
@alexander-akait done. Could we still have the possibility to disable this "opt-in" feature?

**Comment by alexander-akait** [comment]
You can try to disable ESM, using `esModule: false`, also you can try to disable some optimizations (side effects)

**Comment by soluml** [comment]
> @soluml I saw that your global-import-loader replaces https://github.com/fred104/webpack-import-glob-loader in the archetype, how come?

@kevinolivar - A few reasons:

1. It's essentially been abandoned.
2. If I recall correctly, at the time we were using it for our AEM projects, it was causing `npm audit` errors.
3. Aliasing was a big part of our workflow. I even submitted a PR to **webpack-import-glob-loader** but it's just sat there for well over a year now.
4. While certainly not essential for all my projects, I've found it useful in several, especially with AEM projects. I thought I could also improve it by leveraging Enhanced Resolver to not only support aliases but other custom Webpack resolution configurations.

Those are the main ones. The reason I submitted to the archetype was all of the above plus the fact that the company I worked for is a heavy user of the archetype. I wanted aliasing out OOTB and figured a maintained dependency was better than an unmaintained one. If you run into issues with it, let me know and I won't let you hanging for 3+ years, lord willing 😉 

Also, unrelated, thanks for following up and producing a proper example of the issue here. I was in a time crunch when I made this ticket and we were able to find a work-around that satisfied `npm audit` (and the higher ups) so I never had a chance to follow up.
